components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3692,5 +3249,628 @@ subroutine get_time_space_phys_variables( &
+         exit
+      endif !
+   enddo
+
+   qr_present: if (log_qxpresent) then
+
+      dt_left   = dt  !time remaining for sedi over full model (mp) time step
+      prt_accum = 0._rtype  !precip rate for individual category
+
+      !find bottom
+      do k = kbot,k_qxtop,kdir
+         if (qr(k).ge.qsmall) then
+            k_qxbot = k
+            exit
+         endif
+      enddo
+
+      substep_sedi_r: do while (dt_left.gt.1.e-4_rtype)
+
+         Co_max = 0._rtype
+         V_qr = 0._rtype
+         V_nr = 0._rtype
+
+         kloop_sedi_r1: do k = k_qxtop,k_qxbot,-kdir
+
+            qr_notsmall_r1: if (qr_incld(k)>qsmall) then
+
+               !Compute Vq, Vn:
+               nr(k)  = max(nr(k),nsmall)
+               call get_rain_dsd2(qr_incld(k),nr_incld(k),mu_r(k),lamr(k),     &
+                    tmp1,tmp2,rcldm(k))
+               call find_lookupTable_indices_3(dumii,dumjj,dum1,rdumii,rdumjj,inv_dum3, &
+                    mu_r(k),lamr(k))
+               nr(k) = nr_incld(k)*rcldm(k)
+               !mass-weighted fall speed:
+
+               dum1 = vm_table(dumii,dumjj)+(rdumii-real(dumii))*                       &
+                      (vm_table(dumii+1,dumjj)-vm_table(dumii,dumjj))       !at mu_r
+               dum2 = vm_table(dumii,dumjj+1)+(rdumii-real(dumii))*                     &
+                      (vm_table(dumii+1,dumjj+1)-vm_table(dumii,dumjj+1))   !at mu_r+1
+               V_qr(k) = dum1 + (rdumjj-real(dumjj))*(dum2-dum1)         !interpolated
+               V_qr(k) = V_qr(k)*rhofacr(k)               !corrected for air density
+
+               ! number-weighted fall speed:
+
+               dum1 = vn_table(dumii,dumjj)+(rdumii-real(dumii))*                       &
+                      (vn_table(dumii+1,dumjj)-vn_table(dumii,dumjj))       !at mu_r
+               dum2 = vn_table(dumii,dumjj+1)+(rdumii-real(dumii))*                     &
+                      (vn_table(dumii+1,dumjj+1)-vn_table(dumii,dumjj+1))   !at mu_r+1
+
+               V_nr(k) = dum1+(rdumjj-real(dumjj))*(dum2-dum1)            !interpolated
+               V_nr(k) = V_nr(k)*rhofacr(k)                !corrected for air density
+
+            endif qr_notsmall_r1
+
+            Co_max = max(Co_max, V_qr(k)*dt_left*inv_dzq(k))
+            !            Co_max = max(Co_max, max(V_nr(k),V_qr(k))*dt_left*inv_dzq(i,k))
+
+         enddo kloop_sedi_r1
+
+         !-- compute dt_sub
+         tmpint1 = int(Co_max+1._rtype)    !number of substeps remaining if dt_sub were constant
+         dt_sub  = min(dt_left, dt_left/float(tmpint1))
+
+         if (k_qxbot.eq.kbot) then
+            k_temp = k_qxbot
+         else
+            k_temp = k_qxbot-kdir
+         endif
+
+         !-- calculate fluxes ! AaronDonahue, including rflx output
+         do k = k_temp,k_qxtop,kdir
+            flux_qx(k) = V_qr(k)*qr(k)*rho(k)
+            flux_nx(k) = V_nr(k)*nr(k)*rho(k)
+            rflx(k+1) = rflx(k+1) + flux_qx(k) ! AaronDonahue
```

Comment:
I'm not sure why this wasn't an issue before, but the (k+1) broke my test of this PR because if k_temp=kbot then k_temp+1=73, or one more than the total number of layers.  `k+1` should be changed to `k+kdir`.  I can only guess that before his PR `k_temp` never equalled `kbot`???  Either way it is a bug I introduced awhile ago when I included `rflx`.

Reply:
This has been resolved.  It's not a bug, its just that rflx needs to be declared with one more level in the second dimension.

Reply:
Good catch. Thanks, @AaronDonahue .